### PR TITLE
#3291 - Student search

### PR DESCRIPTION
**Pull request by katrina-aot-git**
Student search now contains a SIN column. E2E tests are covered in student.institutions.controller.searchStudents.e2e-spec

Screenshot 2024-07-22 at 1 32 00 PM

### CODE REVIEW

File: sources/packages/web/src/components/common/SearchStudents.vue [hunk 1]
```diff
@@ -72,21 +72,26 @@
     <content-group>
       <toggle-content :toggled="!students?.length">
         <DataTable :value="students">
-          <Column field="firstName" header="First Name" :sortable="true">
+          <Column field="sin" header="SIN" :sortable="true">
+            <template #body="slotProps">
+              {{ slotProps.data.sin }}
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/components/common/SearchStudents.vue:77]: Minor but please use the `sinDisplayFormat`,

**Review by andrewsignori-aot — COMMENTED** [review]
Nice work.
As mentioned, `student.institutions.controller.searchStudents.e2e-spec` is taking care of the Institution E2E tests. At least one E2E test scenario can be added to the Ministry controller.

File: sources/packages/backend/apps/api/src/route-controllers/student/student.aest.controller.searchStudents.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+          id: student.id,
+          firstName: student.user.firstName,
+          lastName: student.user.lastName,
+          birthDate: student.birthDate,
+          sin: student.sinValidation.sin,
+        },
+      ]);
+  });
+
+  it("Should find the student by sin when student.", async () => {
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/api/src/route-controllers/student/student.aest.controller.searchStudents.e2e-spec.ts:76]: The sentence seems to be incomplete: "when student...".

File: sources/packages/backend/apps/api/src/route-controllers/student/student.aest.controller.searchStudents.e2e-spec.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    app = nestApplication;
+    appDataSource = dataSource;
+    studentRepo = dataSource.getRepository(Student);
+    applicationRepo = dataSource.getRepository(Application);
+    collegeFLocation = createFakeInstitutionLocation({ institution: collegeF });
+  });
+
+  it("Should find the student by part of first name.", async () => {
+    // Arrange
+    const { student } = await saveStudentWithApplicationForCollegeF(
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/api/src/route-controllers/student/student.aest.controller.searchStudents.e2e-spec.ts:43]: No need to save a Student associated with an application. Ministry has full access to any student.
Please remove the application associations, a saveFakeStudent should be enough for this E2E file.

File: sources/packages/backend/apps/api/src/route-controllers/student/student.aest.controller.searchStudents.e2e-spec.ts [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        },
+      ]);
+  });
+
+  /**
+   * Saves a student and an application for the student for College F.
+   * @param applicationStatus application status.
+   * @returns an object with student and application persisted in the database.
+   */
+  const saveStudentWithApplicationForCollegeF = async (
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/api/src/route-controllers/student/student.aest.controller.searchStudents.e2e-spec.ts:113]: As per the below comment, this method would not be needed and can be misleading.
https://github.com/bcgov/SIMS/pull/3548/files#r1688715121

File: sources/packages/backend/apps/api/src/route-controllers/student/student.aest.controller.searchStudents.e2e-spec.ts [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+      lastName: "",
+      sin: "",
+    };
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send(searchPayload)
+      .auth(
+        await getAESTToken(AESTGroups.BusinessAdministrators),
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/api/src/route-controllers/student/student.aest.controller.searchStudents.e2e-spec.ts:61]: Minor, expect for old tests, the token is usually acquired during the arrange.

**Review by andrewsignori-aot — COMMENTED** [review]
Thanks for adding the E2E tests. Please take a look at the comments.

File: sources/packages/backend/apps/api/src/route-controllers/student/student.aest.controller.searchStudents.e2e-spec.ts (hunk 5 above)
Comment (katrina-aot-git) [reply sources/packages/backend/apps/api/src/route-controllers/student/student.aest.controller.searchStudents.e2e-spec.ts]: updated

File: sources/packages/backend/apps/api/src/route-controllers/student/student.aest.controller.searchStudents.e2e-spec.ts (hunk 4 above)
Comment (katrina-aot-git) [reply sources/packages/backend/apps/api/src/route-controllers/student/student.aest.controller.searchStudents.e2e-spec.ts]: updated

File: sources/packages/backend/apps/api/src/route-controllers/student/student.aest.controller.searchStudents.e2e-spec.ts (hunk 3 above)
Comment (katrina-aot-git) [reply sources/packages/backend/apps/api/src/route-controllers/student/student.aest.controller.searchStudents.e2e-spec.ts]: updated

File: sources/packages/backend/apps/api/src/route-controllers/student/student.aest.controller.searchStudents.e2e-spec.ts (hunk 2 above)
Comment (katrina-aot-git) [reply sources/packages/backend/apps/api/src/route-controllers/student/student.aest.controller.searchStudents.e2e-spec.ts]: updated

File: sources/packages/backend/apps/api/src/route-controllers/student/student.aest.controller.searchStudents.e2e-spec.ts [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+  getAESTToken,
+} from "../../testHelpers";
+
+describe("StudentMinistryController(e2e)-searchStudents", () => {
+  let app: INestApplication;
+  let appDataSource: DataSource;
+  let collegeF: Institution;
+  let collegeFLocation: InstitutionLocation;
+  let studentRepo: Repository<Student>;
+  let applicationRepo: Repository<Application>;
```
[the remark this answers is not in the document]
Comment (katrina-aot-git) [reply sources/packages/backend/apps/api/src/route-controllers/student/student.aest.controller.searchStudents.e2e-spec.ts]: updated

File: sources/packages/web/src/components/common/SearchStudents.vue (hunk 1 above)
Comment (katrina-aot-git) [reply sources/packages/web/src/components/common/SearchStudents.vue]: updated

**Review by andrewsignori-aot — APPROVED** [review]
Thanks for making the changes, looks good 👍

**Pull request merged**